include/clang/CrossTU/CrossTranslationUnit.h

```diff
@@ -160,6 +160,16 @@ class CrossTranslationUnitContext {
   /// Emit diagnostics for the user for potential configuration errors.
   void emitCrossTUDiagnostics(const IndexError &IE);
 
+  /// Determine the original source location in the original TU for an
+  /// imported source location.
+  /// \p ToLoc Source location in the imported-to AST.
+  /// \p FromLoc Returned source location in the imported-from AST.
+  /// \return The ASTUnit for the source file of FromLoc. If the ToLoc is not an
+  /// imported location or other error occurs, nullptr is returned.
+  clang::ASTUnit *
+  GetImportedFromSourceLocation(const clang::SourceLocation &ToLoc,
```

Comment:
I think, using Expected could be better for error handling and for code comprehension:
`Expected<tuple<ASTUnit, SourceLocation /*FromLoc*/>> GetImportedFromSourceLocation(const clang::SourceLocation &ToLoc)`